src/common.rs

```diff
@@ -61,73 +62,53 @@ pub fn verify_filename(filename: String) -> Result<String, io::Error> {
 }
 
 /// compresses abs_file_or_dirname into system_tmp_dir/file_or_dirname.gz if a file or system_tmp_dir/file_or_dirname.tar.gz if a dir
-pub fn compress(
+pub async fn compress(
```

Comment:
you **cannot** use blocking io in async things. so lots more things need to be async here.